src/optimize.jl

```diff
@@ -17,6 +17,9 @@ function optimize!(gp::GPBase; method = LBFGS(), domean::Bool = true, kern::Bool
                    noise::Bool = true, lik::Bool = true, kwargs...)
     params_kwargs = get_params_kwargs(gp; domean=domean, kern=kern, noise=noise, lik=lik)
     # println(params_kwargs)
+    if !is_data_updated(gp) 
```

Comment:
If I understand this correctly, you want to avoid calling KernelData() every time a new point is added, but obtain it if the user wants to refit kernel parameters after adding a few points. We should try to think of a better way. Some possibilities:
1. implement elastic KernelData types, but that's a lot of work
2. use multiple dispatch: if P<:ElasticPDMat, a call to optimize! starts by recomputing the KernelData.
3. implement a more general way to invalidate cached objects, like KernelData, but also mll, alpha, and dmll

Reply:
I think option 1. would be nice in the long term.
For option 2: Did you think of a pattern like 
```julia
optimize(gp) = _optimize(gp)
optimize(gp::GPE{elastic}) = begin updatekerneldata(gp); _optimize(gp) end
```
where `_optimize` is the old `optimize` function?

Reply:
Yes. The only problem is that it's not very pretty, but I think it's better than the `is_data_updated` solution. What do you think?